drivers/clock_control/clock_control_esp32.c

```diff
@@ -655,11 +675,18 @@ static int clock_control_esp32_init(const struct device *dev)
 {
 	const struct esp32_clock_config *cfg = dev->config;
 	bool ret;
-#if !defined(CONFIG_SOC_SERIES_ESP32C6)
 	soc_reset_reason_t rst_reas;
-	rtc_config_t rtc_cfg = RTC_CONFIG_DEFAULT();
 
 	rst_reas = esp_rom_get_reset_reason(0);
+
+#if defined(CONFIG_SOC_SERIES_ESP32C6)
```

Comment:
this if/endif structure here is starting to be a mess. Could we make it more readable (and maintainable)?